doc/ntp/ntp-design.md

```diff
@@ -0,0 +1,898 @@
+                    must "(current() != 'mgmt') or (/mvrf:sonic-mgmt_vrf/mvrf:MGMT_VRF_CONFIG/mvrf:vrf_global/mvrf:mgmtVrfEnabled = 'true')" {
+                        error-message "Must condition not satisfied. Try enable Management VRF.";
+                    }
+
+                    type string {
+                        pattern "mgmt|default";
+                    }
+
+                    description
+                        "NTP can be enabled only in one VRF at time. In this revision, it is either default VRF or Management VRF.";
+                }
+
+                leaf state {
+                    type state;
+                    default enabled;
+                    description "NTP feature state";
+                }
+
+                leaf authentication {
+                    type state;
+                    default disabled;
+                    description "NTP authentication state";
+                }
+
+                leaf server_role {
+                    type state;
+                    default enabled;
+                    description "NTP server functionality state";
+                }
+
+            } /* end of container global */
+
+        } /* end of container NTP */
+
+        container NTP_SERVER {
+
+            description "NTP SERVER part of config_db.json";
+
+            list NTP_SERVER_LIST {
+                max-elements 10;
+
+                key "server_address";
+
+                leaf server_address {
+                    type inet:host;
+                }
+
+                leaf state {
+                    type state;
+                    default enabled;
+                    description "NTP server state";
+                }
+
+                leaf key {
+                    type uint16;
+                    description "NTP server key";
+                }
+
+                leaf trusted {
+                    type yes-no;
+                    default no;
+                }
+
+                leaf version {
+                    type uint8 {
+                        range "3..4" {
+                            error-message "Failed NTP version";
+                        }
+                    }
+                    default 4;
+                }
+
+                leaf association-type {
+                    type association-type;
+                    default server;
+                }
+
+                leaf iburst {
+                    type on-off;
+                    default off;
+                }
+            } /* end of list NTP_SERVER_LIST */
+
+        } /* end of container NTP_SERVER */
+
+        container NTP_KEY {
+
+            description "NTP authentication keys configuration";
+
+            list NTP_KEYS_LIST {
+                description "TBD";
+                key "id";
+
+                leaf id {
+                    type uint16;
+                }
+
+                leaf type {
+                    type ntp-key-type;
+                }
+
+                leaf value {
```

Comment:
This should be an encrypted key, please add a description accordingly.

Reply:
will do